Testing/SystemTests/tests/analysis/INTERReductionTest.py

```diff
@@ -0,0 +1,349 @@
+        'Name': ['Theta', 'ScaleFactor', 'AirSLD', 'BulkSLD', 'Roughness', 'BackGround', 'Resolution',
+                 'SLD_Layer0', 'd_Layer0', 'Rough_Layer0', 'Cost function value'],
+        'Value': [2.2999999999999998, 1, 0, 0, 0, 7.9488100000033575e-06, 5, 8.2865399998548345e-07,
+                  31.315399998680391, 0, 0.90910656437440196],
+        'Error': [0, 0, 0, 0, 0, 1.1949723340559929e-07, 0, 2.2338118077681473e-08, 0.9921668942754267, 0, 0]}
+    expected_fit_covariance={
+        'Name': ['BackGround', 'SLD_Layer0', 'd_Layer0'],
+        'BackGround':[100, -64.519943050096259, 60.825620358818924],
+        'SLD_Layer0':[-64.519943050096032, 100, -96.330851077988683],
+        'd_Layer0': [60.825555558936458, -96.33084065170361, 100]}
+
+    def __init__(self):
+        super(INTERReductionTest, self).__init__()
+        self.tolerance = 1e-6
+
+    def requiredFiles(self):
+        return [self.reference_result_file, self.runs_file]
+
+    def validate(self):
+        return (self.result_workspace, self.reference_result_file)
+
+    def runTest(self):
+        SetupInstrument()
+        Load(self.runs_file, OutputWorkspace=self.runs_workspace)
+        workspaces_to_exclude_from_result = AnalysisDataService.Instance().getObjectNames()
+        CreateTransmissionWorkspaces(self.first_transmission_run_names,
+                                     self.second_transmission_run_names,
+                                     self.transmission_workspace_names)
+
+        TestEventDataTimeSlicing(self.event_run_numbers)
+        TestReductionOfThreeAngleFringedSolidLiquidExample([45222,45223,45224])
+        TestReductionOfTwoAngleAirLiquidExample([44984,44985])
+        TestFittingOfReducedData(44990,44991,self.expected_fit_params,self.expected_fit_covariance)
+
+        RemoveWorkspaces(workspaces_to_exclude_from_result)
+        GroupWorkspaces(InputWorkspaces=AnalysisDataService.Instance().getObjectNames(),
+                        OutputWorkspace=self.result_workspace)
+        mtd[self.result_workspace].sortByName()
+
+    @staticmethod
+    def regenerateRunsFile():
+        SetupInstrument()
+        RegenerateRunsFile(INTERReductionTest.first_transmission_run_names +
+                           INTERReductionTest.second_transmission_run_names,
+                           INTERReductionTest.run_numbers,
+                           INTERReductionTest.event_run_numbers)
+
+    @staticmethod
+    def regenerateReferenceFileByReducing():
+        SetupInstrument()
+        test = INTERReductionTest()
+        test.runTest()
+        SaveNexus(InputWorkspace=INTERReductionTest.result_workspace,
+                  Filename=INTERReductionTest.reference_result_file)
+
+    @staticmethod
+    def regenerateReferenceFileFromDirectory(reference_file_directory):
+        SetupInstrument()
+        RegenerateReferenceFile(reference_file_directory, INTERReductionTest.reference_result_file)
+
+
+def SetupInstrument():
```

Comment:
Use `camalCase` (more "Mantid-ish") or `snake_case` (more "Python-ish") for function names.